### PR TITLE
CORDA-1238: Updated JacksonSupport to support SerializedBytes, CertPath, X509Certificate and the signature classes

### PR DESCRIPTION
**Pull request by shamsasari**
SerializedBytes are first converted to the object it represents before being serialised as a pojo.

These changes will be needed to support the the blob inspector when it will output to YAML/JSON.

### CODE REVIEW

**Comment by shamsasari** [comment]
I've made extension use of mixins as it's very good at reducing boilerplate.

**Comment by shamsasari** [comment]
Those changes were made in https://github.com/corda/corda/pull/3102

**Comment by tudor-malene** [comment]
Right, but still, the api file does not reflect the code at the moment.

**Comment by shamsasari** [comment]
The API file is not meant to be updated when new API is added, so the fact that `nodeInfoFromParty` is not listed is expected. It will be added when we do the next release and the file is re-generated.

As for the missing deprecation that's something for @anthonykeenan or @chrisr3 to look at if it's an issue. Otherwise, it's not an issue with the PR itself.

File: client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt [hunk 1]
```diff
@@ -90,25 +95,26 @@ object JacksonSupport {
...
-            addDeserializer(CompositeKey::class.java, CompositeKeyDeseriaizer)
-            addSerAndDeser(toStringSerializer, NetworkHostAndPortDeserializer)
-            // TODO Add deserialization which follows the same lookup logic as Party
-            addSerializer(PartyAndCertificate::class.java, PartyAndCertificateSerializer)
-            addDeserializer(NodeInfo::class.java, NodeInfoDeserializer)
-
-            listOf(TransactionSignatureSerde, SignedTransactionSerde).forEach { serde -> serde.applyTo(this) }
-
-            // Using mixins to fine-tune the default serialised output
+            setMixInAnnotation(BigDecimal::class.java, BigDecimalMixin::class.java)
```
Comment (tudor-malene) [inline client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt]: Maybe worth investigating ( as part of some other PR ?) why we need special behaviour for BigDecimals.

**Pull request merged**